extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js

```diff
@@ -174,6 +175,57 @@ export class Criteria {
   }
 }
 
+/**
+ * Regex for the width-selection portion of a srcset, so for the
+ * general grammar: (URL [NUM[w|x]],)*, this should express "NUMw".
+ * E.g. in "image1.png 100w, image2.png 50w", this matches "100w" and "50w"
+ */
+const srcsetWidthRe = /\s+([0-9]+)w(,|$)/g;
+
+/**
+ * Parses srcset partially to get the maximum defined intrinsic width.
+ * @param {!Element} img
+ * @return {number} -1 if no srcset, or if srcset is defined by dpr instead of
```

Comment:
Clever using `-1` here as no `srcset` value in that case, but perhaps you could explain why (I'm assuming it's the `maxWidthFromSrcset > naturalWidth)` check below.

Reply:
Good point!